### DISCUSSION TITLE
`-stats` flag does not return `100%` when Scan is Complete

### DISCUSSION THREAD

**Discussion by oliv10** [projectdiscovery/nuclei#5134] [Q&A]
# Edit:
### Nuclei version:
3.2.5

### Current Behavior:

When using the stats flag the last output it puts is not at 100% nor does the requests made equal the total number of requests. This does not change whether using the -j flag output or not.

### Expected Behavior:

It should give the proper data. 100% completed when done as the final output.

### Steps To Reproduce:

1. Run `nuclei -u http://honey.scanme.sh -stats -silent`
2. See final stats output is not 100%

### Anything else:
Possibly Related to #4387 

<img width="765" alt="Screenshot 2024-05-01 at 11 56 07 AM" src="https://github.com/projectdiscovery/nuclei/assets/18117030/d0792450-fd2b-4df2-b88b-f95f69adcde6">

<img width="1359" alt="Screenshot 2024-05-01 at 12 00 49 PM" src="https://github.com/projectdiscovery/nuclei/assets/18117030/5a255909-d65e-4487-913c-9fe9c2ad01e7">

Final 3 stats reports.

**Comment by ehsandeep** [comment]
@oliv10 stats shows approx data only.

**Reply by oliv10** [reply]
@ehsandeep I agree, it is approximate data _**when the scan is running**_. When the scan has finished to completion it should return 100%. If for some reason the scan cannot complete fully it should put an error message. Or why not add a bool value completed in the stats output.
